packages/solutions/app-tools/src/old.ts

```diff
@@ -0,0 +1,193 @@
+          ? 'rspack'
+          : 'webpack',
+    }),
+    serverBuildPlugin(),
+    deployPlugin(),
+  ],
+
+  setup: api => {
+    const appContext = api.useAppContext();
+    api.setAppContext({
+      ...appContext,
+      toolsType: 'app-tools',
+    });
+
+    const locale = getLocaleLanguage();
+    i18n.changeLanguage({ locale });
+
+    return {
+      async beforeConfig() {
+        const userConfig = api.useConfigContext();
+        const appContext = api.useAppContext();
+        if (userConfig.output?.tempDir) {
+          api.setAppContext({
+            ...appContext,
+            internalDirectory: path.resolve(
+              appContext.appDirectory,
+              userConfig.output.tempDir,
+            ),
+          });
+        }
+      },
+      async commands({ program }) {
+        await devCommand(program, api);
+        await buildCommand(program, api);
+        serverCommand(program, api);
+        deployCommand(program, api);
+        newCommand(program, locale);
+        inspectCommand(program, api);
+        upgradeCommand(program);
+        deprecatedCommands(program);
+      },
+
+      async prepare() {
+        const command = getCommand();
+        if (command === 'deploy') {
+          const isSkipBuild = ['-s', '--skip-build'].some(tag => {
+            return getArgv().includes(tag);
+          });
+          // if skip build, do not clean dist path
+          if (isSkipBuild) {
+            return;
+          }
+        }
+
+        // clean dist path before building
+        if (
+          command === 'dev' ||
+          command === 'start' ||
+          command === 'build' ||
+          command === 'deploy'
+        ) {
+          const resolvedConfig = api.useResolvedConfigContext();
+          if (resolvedConfig.output.cleanDistPath) {
+            const appContext = api.useAppContext();
+            await emptyDir(appContext.distDirectory);
+          }
+        }
+      },
+
+      async watchFiles() {
+        const appContext = api.useAppContext();
+        const config = api.useResolvedConfigContext();
+        const files = await generateWatchFiles(
+          appContext,
+          config.source.configDir,
+        );
+
+        const watchFiles = castArray(config.dev.watchFiles);
+        watchFiles.forEach(({ type, paths }) => {
+          if (type === 'reload-server') {
+            files.push(...(Array.isArray(paths) ? paths : [paths]));
+          }
+        });
+
+        return files;
+      },
+
+      // 这里会被 core/initWatcher 监听的文件变动触发，如果是 src 目录下的文件变动，则不做 restart
+      async fileChange(e: {
+        filename: string;
+        eventType: string;
+        isPrivate: boolean;
+      }) {
+        const { filename, eventType, isPrivate } = e;
+
+        if (!isPrivate && (eventType === 'change' || eventType === 'unlink')) {
+          const { closeServer } = await import('./utils/createServer.js');
+          await closeServer();
+          await restart(api.useHookRunners(), filename);
```

Comment:
restart will invoke cli.init, we should migrate to new cli instance, now it's in `@modern-js/core`